packages/components/src/components/Graph/Node.js

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/* istanbul ignore file */
+import React, { PureComponent } from 'react';
+
+import CheckmarkFilled from '@carbon/icons-react/lib/checkmark--filled/16';
+import ChevronDown from '@carbon/icons-react/lib/chevron--down/16';
+import CloseFilled from '@carbon/icons-react/lib/close--filled/16';
+import Undefined from '@carbon/icons-react/lib/undefined/16';
+
+import InlineLoading from './InlineLoading';
+import './Graph.scss';
+
+export default class Node extends PureComponent {
+  render() {
+    const { children, height, status, type, width, label = type } = this.props;
+
+    let StatusIcon;
+    if (type === 'Task' || status === 'error') {
+      switch (status) {
+        case 'error':
```

Comment:
This reads a little strange, make sense to move state === error to its own conditional above ?

Reply:
I had that originally but didn't like the duplication and potential for inconsistency in the icons used, we can have a Task with status 'error' or a Step with status 'error'. Tasks always show an icon, Steps only show icon on error.

Open to any suggestions to clean this up.